### PR TITLE
Feature/add parallel gateway

### PR DESCRIPTION
**Pull request by ElRaptorus**
- Add sample for running a parallel gateway
- Add test and suite for running a test for that process

### CODE REVIEW

File: ParallelGateway/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# ParallelGateway
+
+## How to use it
+
+- Checkout the Branch
+- Go into your Process Engine skeleton and start the database
+- Go into the `ParallelGateway` folder
+- ```npm install && npm run build && npm start```
+- Open `localhost:9000` and start the `ParallelGateway` process
```
Comment (LeoTT) [inline ParallelGateway/README.md]: - User also needs to set up bpmn-studio and start it.
- Would add links to the repos of skeleton and bpmn-studio.
- Should be `localhost:8080`, right? Only 9000 when you're using au, but `npm start`, which is the correct way, uses port 8080.

Comment (ElRaptorus) [reply ParallelGateway/README.md]: You are correct (on all accounts), thx for pointing this out.

**Comment by ElRaptorus** [comment]
@LeoTT done

File: ParallelGateway/config/development/feature/feature_service.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "skeleton_server": {
+      "features": {
+        "Name": "Skeleton_Server",
+        "Environment": "Server",
+        "Platform": "NodeJS",
+        "PlatformVersion": "7.6.0",
+        "UI": true
+      }
+    },
+    "skeleton_pi": {
```
Comment (Paulomart) [inline ParallelGateway/config/development/feature/feature_service.json]: Please check if we need this

File: ParallelGateway/config/development/security/security_service.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+
+}
```
Comment (Paulomart) [inline ParallelGateway/config/development/security/security_service.json]: insert missing newline

File: ParallelGateway/config/development/services/mail.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (Paulomart) [inline ParallelGateway/config/development/services/mail.json]: We dont even register a mail service in the ioc container; I doubt we need this here

File: ParallelGateway/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+After that, you can use the sample by doing the following:
+- Checkout the Branch
+- Go into the `ParallelGateway` folder
+- Run `npm install && npm run build`
+- Run `npm start` to start the backend application 
+
+This sample implements its own [Skeleton Application](), which will add the
+`ParallelGateway` BPMN to your local database.
+So after setting up your BPMN studio and installting, building and starting
```
Comment (LeoTT) [inline ParallelGateway/README.md]: typo: `installting` -> `installing`

File: ParallelGateway/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+the `ParallelGateway` app, you are good to go.
+You can now open `localhost:8080` and start the `ParallelGateway` process.
+
+## What it does
+
+It runs five exection paths that run in parallel to each other, each simulating
```
Comment (LeoTT) [inline ParallelGateway/README.md]: typo: `exection` -> `execution`

File: ParallelGateway/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# ParallelGateway
+
+## How to use it
+
+Before you can use this sample, you need to setup and start an instance of
+the [BPMN Studio](https://github.com/process-engine/bpmn-studio).
```
Comment (LeoTT) [inline ParallelGateway/README.md]: The user also needs to install postgres or docker and run the database on the right port.
## Requirements
-  [BPMN Studio](https://github.com/process-engine/bpmn-studio) installed and running
- [database](https://github.com/process-engine/skeleton/tree/develop/database) for the ProcessEngine installed and running

File: ParallelGateway/config/development/services/mail.json (hunk 4 above)
Comment (ElRaptorus) [reply ParallelGateway/config/development/services/mail.json]: Yes we do, through: `'@essential-projects/services'`, see:
https://github.com/process-engine/example_processes/blob/c50d61feb290584e138ed897fb8b48fc7a1f9f7a/ParallelGateway/test/setup.js#L8

This package contains a number of services besides the mail service, so I'd like to keep this here.

**Comment by ElRaptorus** [comment]
@LeoTT @Paulomart done

**Pull request merged**